internal/state/registry.go

```diff
@@ -29,6 +30,18 @@ func sFnRegistryConfiguration(ctx context.Context, r *reconciler, s *systemState
 		setK3dRegistry(s)
 	}
 
+	if condition, ok := s.instance.GetCondition(v1alpha1.ConditionTypeConfigured); ok {
+		if condition.Status == metav1.ConditionFalse {
```

Comment:
I would suggest using [this](https://pkg.go.dev/k8s.io/apimachinery@v0.25.4/pkg/api/meta#IsStatusConditionFalse) method. Would be much easier to use/edit and more readable.

Reply:
And one more thing. I think we should set the condition to True if it's not equal to True. Not only if it's False ( in this case we would omit use case when a condition is in the Unknown state )